engines/prover.h

```diff
@@ -27,8 +27,11 @@ namespace pono {
 class Prover
 {
  public:
-  Prover(const Property & p, smt::SmtSolver & s);
-  Prover(const PonoOptions & opt, const Property & p, smt::SmtSolver & s);
+  Prover(const Property & p, smt::SolverEnum se);
+  Prover(const Property & p, smt::SmtSolver s);
+  Prover(const PonoOptions & opt, const Property & p, smt::SolverEnum se);
+  Prover(const PonoOptions & opt, const Property & p, smt::SmtSolver s);
```

Comment:
const ref to smtsolver

Reply:
Thanks, addressed with 0dbc624d1d5d9ab57c7977b758211620fdff5ecd